feat: DIA-825: Setup Model list page placeholder

nick-skriabin opened the pull request:
This PR introduces the models list page as the part of the PW.

Consider this as a placeholder page for the rest of the PW functionality.

### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated
- [ ] Docs have been added/updated
- [X] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [X] Self-reviewed and ran all changes on a local instance

### What feature flags were used to cover this change?
`fflag_feat_all_dia_835_prompter_workflow_long`

### What alternative approaches were there?
None

### This change affects (describe how if yes)
- [ ] Performance
- [ ] Security
- [X] UX

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [X] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [ ] e2e (codecept)
- [ ] integration (cypress)
- [ ] unit (jest)

### Which logical domain(s) does this change affect?
Prompter workflow

nick-skriabin committed:
Models page

nick-skriabin committed:
Add feature flag

nick-skriabin committed:
Add menubar icon

hlomzik commented on label_studio/organizations/urls.py:
```diff
@@ -3,36 +3,48 @@
 from organizations import api, views
 
-app_name = 'organizations'
+app_name = "organizations"
```
as I can see the rest of python code uses single quotes, so that can mess with their linting

nick-skriabin replied:
fixed

hlomzik commented on label_studio/organizations/urls.py:
```diff
@@ -3,36 +3,48 @@
+    ),
+    path("people/", include(_urlpatterns)),
+    # TODO: temporary route, remove as needed
+    path("prompt/", views.simple_view, name="prompter"),
```
do we really need to add all urls to python routing? I thought it's handled by react 😭

nick-skriabin replied:
there must be a route on the backend, unfortunately. otherwise reloading the page will cause 404

hlomzik replied:
but that's strange and not how modern apps work... known backend routes and statics should be handled via nginx or smth, the rest will go to frontend to apply routing. why didn't we choose this way?

hlomzik commented on web/apps/labelstudio/src/assets/images/heidi-ai.svg:
love it!!

hlomzik commented on web/apps/labelstudio/src/components/Breadcrumbs/Breadcrumbs.js:
```diff
@@ -24,6 +24,8 @@ export const Breadcrumbs = () => {
     }
   }, [reactBreadcrumbs, config]);
 
+  console.log(breadcrumbs);
```
reminder in case you missed it

hlomzik commented on web/apps/labelstudio/src/pages/Organization/Prompter/@components/EmptyList.tsx:
what's with the `@`? is this new approach to something?

nick-skriabin replied:
it tells you that it's local components directoy for the specific page instead of general one. we used to mix these which is weird. also useful when using import autocompletion. even if there are multiple components with the same name, with `@` you'll see that it's page-specific and not intended for global use

I will be documenting this stuff along with UI lib docs

hlomzik commented on web/webpack.config.js:
```diff
@@ -145,19 +146,26 @@ module.exports = composePlugins(withNx({
             localIdentName: css_prefix + '[local]', // Customize this format
           };
         }
+        console.log(_r, cssLoader)
```
another one. plus you don't use `log` you imported earlier

nick-skriabin replied:
yeah, gotta cover that under the linting strategy initiative

nick-skriabin committed:
Remove console.log

nick-skriabin committed:
Fix python linting

nick-skriabin committed:
Add color variables

nick-skriabin committed:
Merge branch 'develop' into fb-dia-825/models-list-page

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7742127913

nick-skriabin committed:
Rename "prompt/" to "models/"

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7742853911
